pkg/logfilter/logfilter.go

```diff
@@ -0,0 +1,77 @@
+package logfilter
+
+import (
+	"net/url"
+	"strings"
+)
+
+type Filter interface {
+	Allow(string)
+	IsAllowed(string) bool
+	FilterURL(url *url.URL) *url.URL
+	FilterURLString(url string) string
+	FilterQueryParams(params url.Values) url.Values
+}
+
+type mapKey struct{}
+
+type logFilter struct {
+	allowedMap map[string]mapKey
+}
+
+var _ Filter = &logFilter{}
+
+func New() Filter {
+	return &logFilter{allowedMap: make(map[string]mapKey)}
+}
+
+func (l *logFilter) Allow(key string) {
+	l.allowedMap[strings.ToLower(key)] = mapKey{}
+}
+
+func (l *logFilter) IsAllowed(key string) bool {
+	if _, ok := l.allowedMap[strings.ToLower(key)]; ok {
+		return true
+	}
+
+	return false
+}
+
+func (l *logFilter) FilterURLString(urlString string) string {
+	parsedUrl, err := url.Parse(urlString)
+
+	if err != nil {
+		return urlString
```

Comment:
If the URL fails to parse, perhaps it makes sense to return a magic value instead of the original URL just so invalid URL that make it here never leak anything.

Something like:

```suggestion
		return "FILTEREDINVALIDURL"
```

Reply:
Good call, I'll add that and a relevant test 👍